### PR TITLE
Add hmmer/eslreformat

### PR DESCRIPTION
**Pull request by erikrikarddaniel**
## PR checklist

The `esl-reformat` tool distributed with HMMER reformats aligned or not sequence files between different formats.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by erikrikarddaniel** [comment]
Did I do something so that two of the tests were skipped?

**Comment by nvnieuwk** [comment]
It seems like `tests/config/pytest_modules.yml` was removed. I think that's the problem

**Comment by erikrikarddaniel** [comment]
> It seems like `tests/config/pytest_modules.yml` was removed. I think that's the problem

Thanks. `prettier tests/config/pytest_modules.yml` reports lots of errors and outputs an empty file. I must be doing something wrong...

**Comment by nvnieuwk** [comment]
> 

You can run `prettier -w .`. This should fix this normally (this command checks for linting errors and automatically fixes those it can fix)

File: modules/hmmer/eslreformat/main.nf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::hmmer=3.3.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/hmmer:3.3.2--h1b792b2_1':
+        'quay.io/biocontainers/hmmer:3.3.2--h1b792b2_1' }"
+
+    input:
+    tuple val(meta), path(seqfile)
+    val   format
```
Comment (nvnieuwk) [inline modules/hmmer/eslreformat/main.nf:12]: Inputs should only be used for actual file inputs. You can specify the `format` in the `nextflow.config` file using `ext.args` like: https://github.com/nf-core/modules/blob/b5e3d16d80abfc6aa1c86ffde8c1b507bea80f0d/tests/modules/bcftools/convert/nextflow.config#L9-L11

File: modules/hmmer/eslreformat/meta.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test' ]
+  - seqfile:
+      type: file
+      description: Sequences, aligned or not, in any supported format
+      pattern: "*"
+  - format:
```
Comment (nvnieuwk) [inline modules/hmmer/eslreformat/meta.yml:24]: Don't forget to remove format here too :)

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

**Comment by erikrikarddaniel** [comment]
> > 
> 
> You can run `prettier -w .`. This should fix this normally (this command checks for linting errors and automatically fixes those it can fix)

Right. I had a duplicate entry, which somehow messed things up. Deleting that seems to have worked.

File: modules/hmmer/eslreformat/main.nf (hunk 1 above)
Comment (erikrikarddaniel) [reply modules/hmmer/eslreformat/main.nf]: OK, makes sense. I thought this was only for "real" params with minuses...

File: modules/hmmer/eslreformat/main.nf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    output:
+    tuple val(meta), path("*.sequences.gz"), emit: seqreformated
+    path "versions.yml"                    , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
```
Comment (nvnieuwk) [inline modules/hmmer/eslreformat/main.nf:22]: You could also add a check here to see if the user has specified the format in the `args` and if not, you could add a default. A bit like this: https://github.com/nf-core/modules/blob/b5e3d16d80abfc6aa1c86ffde8c1b507bea80f0d/modules/bcftools/convert/main.nf#L31-L35

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: modules/hmmer/eslreformat/main.nf (hunk 3 above)
Comment (erikrikarddaniel) [reply modules/hmmer/eslreformat/main.nf:22]: That's a little bit tricky as the available formats depends on whether the input file is aligned or not. I'd rather not have to guess that, as there are so many file suffixes used for various sequence formats.

Another problem with this is that the format must come second last, after all the `--` params, before the filename, so the caller needs to get the order right in the config file.

Comment (nvnieuwk) [reply modules/hmmer/eslreformat/main.nf:22]: Allright, maybe better to leave it out then. But you should maybe specify in `meta.yml` that the user should give a format in `args`

Comment (erikrikarddaniel) [reply modules/hmmer/eslreformat/main.nf:22]: Yes, I was looking for a place to document that, thanks. As it doesn't fit the headers in `meta.yml`, should one do that as a comment? Do you have an example of how arguments are documented in `meta.yml`?

Comment (nvnieuwk) [reply modules/hmmer/eslreformat/main.nf:22]: I don't really know of an example, but I think it's okay if you just add it to the top description. Or you could add it to the description of the output file?

Comment (erikrikarddaniel) [reply modules/hmmer/eslreformat/main.nf:22]: OK, done and pushed, in the hope it didn't get too long!

Comment (nvnieuwk) [reply modules/hmmer/eslreformat/main.nf:22]: Looks good!

**Pull request merged**